contribute/design/1.x/cstor/20200511-cstor-backupandrestore.md

```diff
@@ -0,0 +1,259 @@
+
+##### Velero Server Sends DeleteSnapshot Request
+
+Velero server sends `DeleteSnapshot` API of Velero Interface to delete the backup/snapshot with argument `snapshotID`. This snapshotID is generated during the backup creation of this snapshot. Velero Interface will execute the DeleteSnapshot API of cStor backup/restore module. cStor backup/restore module is responsible for performing below steps:
+1. Delete the PVC object for this backup from the cloud provider.
+2. Execute REST API(`latest/backups/`) of the CVC-Operater to delete the resources created for this particular backup.
+    2.1 Delete the CStorCompleteBackup resources if it is not a complete backup or if that is the only backup exists for that volume(So next backup will be complete backup).
+    2.2 Delete the snapshot created for that backup.
+    2.3 Delete the CStorBackUp resource.
+3. Execute the `RemoveSnapshot` API of the cloud interface to delete the snapshot from the cloud provider.
+
+##### Velero Server Sends CreateVolumeFromSnapshot Request
+
+Velero Server will execute the `CreateVolumeFromSnapshot` API of the velero interface to restore the backup with the argument (snapshotID, volumeType). Velero interface will execute `CreateVolumeFromSnapshot` API of cStor backup/restore module. cStor backup/restore module will perform following below steps:
+1. Download the PVC object from the cloud provider through a cloud interface and deploy the PVC. If PVC already exists in the cluster then skip the PVC creation(only for remote restore).
+2. Check If PVC status is bounded.
+3. Execute the REST API(`latest/restore`) with CVC-Operator restore details(includes the IP address of snapshot receiver/sender module) to initiate the restore of the snapshot.
+   3.1 Create CVC if it doesn't exist in the cluster.
+   3.2 Wait till CVC comes to Bound state.
+   3.3 Create a replica count number of restore CR’s which will be responsible for dumping backup data into the volume dataset.
+4. Call cloud interface API `RestoreSnapshot` to download the snapshot from the cloud provider.
+
+##### Accessing CVC REST Endpoint
+
+- One can access the REST endpoint of CVC by fetching the details of CVC-Operator service. Below command will be help to get the CVC-Operator service
+  kubectl get service -n <openebs_namespace> -l openebs.io/component-name: cvc-operator-svc
+
+##### Velero Server Sends DeleteSnapshot Request
+
+Restore delete will delete restore resource object onlu.
+Note: It will not delete the resources restored in that restore(ex: PVC).
+
+##### Work flow in backup controller
+
+When REST API `/latest/backups` is executed it creates CStorBackUp CR. Backup controller which present in pool-manager will get event and perform the following operations:
+1. Update the Status of CStorBackUp resource as `InProgress`(Which will help to understand the backup process is started).
```

Comment:
existing maya-apiserver flow is : 
- endpoint will create cstorbackup with status `Pending`
- pool controller will update status to `Init`
- pool controller will process this cr in next sync and update its status to `InProgress`

Reply:
same for restore

Reply:
updated PTAL